File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/items/TitleProperty.java
package com.mdgd.pokemon.ui.pokemon.items;

public class TitleProperty implements PokemonProperty {

    private final int titleResId;
    private final int nestingLevel;

    public TitleProperty(int titleResId) {
        this(titleResId, 0);
    }

    public TitleProperty(int titleResId, int nestingLevel) {
        this.titleResId = titleResId;
        this.nestingLevel = nestingLevel;
    }

    public int getTitleResId() {
        return titleResId;
    }

    @Override
    public int getViewHolderType(int position) {
        return PROPERTY_TITLE;
    }

    @Override
    public int getNestingLevel() {
        return nestingLevel;
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/dto/TitleProperty.kt
package com.mdgd.pokemon.ui.pokemon.dto

open class TitleProperty @JvmOverloads constructor(
    val titleResId: Int, override val nestingLevel: Int = 0
) : PokemonProperty {
    override fun getViewHolderType(position: Int): Int = PokemonProperty.PROPERTY_TITLE
}
